.changeset/fifty-apricots-count.md

```diff
@@ -0,0 +1,6 @@
+---
+"@comet/admin": patch
+---
+
+Fix a bug in `FinalForm` where the submit mutation wasn't correctly awaited if a `FinalForm` inside an `EditDialog` used an asynchronous validation. 
```

Comment:
I don't think this is specific to EditDialog, but a bug in our FinalForm component in general?

Reply:
I made the message more specific. What do you think?

Reply:
It's better, but IMO the title is too long. Maybe some parts of it can be moved into the description

Maybe something like "Fix saveAction in RouterPrompt inside a FinalForm" as title, and the remainder as description?